tests/test_validation.py

```diff
@@ -140,3 +140,29 @@ def test_parameter_validator(monkeypatch):
     with pytest.raises(BadRequestProblem) as exc:
         validator.validate_request(request)
     assert exc.value.detail.startswith("'x' is not one of ['a', 'b']")
+
+
+async def test_stream_replay():
+    messages = [
+        {"body": b"message 1", "more_body": True},
+        {"body": b"message 2", "more_body": False},
+    ]
+
+    async def receive():
+        return b""
+
+    wrapped_receive = AbstractRequestBodyValidator._insert_messages(
+        receive, messages=messages
+    )
+
+    replay = []
+    more_body = True
+    while more_body:
+        message = await wrapped_receive()
+        replay.append(message)
+        more_body = message.get("more_body", False)
+
+        if len(replay) > len(messages):
```

Comment:
Why is this check needed?

Reply:
If the first message is always returned as before, we get stuck in this while loop. Updated this to an assert statement with some more info.